src/deadline/job_attachments/models.py

```diff
@@ -230,9 +230,25 @@ def from_root_path(root_path: str) -> JobAttachmentS3Settings:
 
         return JobAttachmentS3Settings(path_split[0], "/".join(path_split[1:]))
 
+    @staticmethod
+    def from_s3_root_uri(uri: str) -> JobAttachmentS3Settings:
+
+        try:
+            bucket, key = uri.replace("s3://", "").split("/", maxsplit=1)
```

Comment:
Nit, urllib.parse will make this code work better:
https://docs.python.org/3/library/urllib.parse.html

```
ParseResult(scheme='scheme', netloc='netloc', path='/path;parameters', params='',
            query='query', fragment='fragment')
o = urlparse("http://docs.python.org:80/3/library/urllib.parse.html?"
             "highlight=params#url-parsing")
o
ParseResult(scheme='http', netloc='docs.python.org:80',
            path='/3/library/urllib.parse.html', params='',
            query='highlight=params', fragment='url-parsing')
```